[linux-6.6.y] Add support for Zhaoxin Processors

leoliu-oc opened the pull request:
Let Linux supports new Zhaoxin CPU features and " Shanghai " CPU vendor ID;
let related drivers supports new Zhaoxin CPU FMS.
Fix a CPU topo limitation of Installed kernel use CPUID B leaf to get CPU topo as default,
add 4leaf get CPU cacheinfo additionally.

Note：
Support for the Zhaoxin processor includes two vendor IDs, zhaoxin and centaur, so we will synchronously modify centaur.c and zhaoxin.c.

Include:

1. x86/cpufeatures: Add Zhaoxin feature bits
2. x86/cpu: Add detect extended topology for Zhaoxin CPUs

opsiff commented on arch/x86/include/asm/cpufeatures.h:
```diff
@@ -148,6 +148,8 @@
 /* VIA/Cyrix/Centaur-defined CPU features, CPUID level 0xC0000001, word 5 */
 #define X86_FEATURE_XSTORE		( 5*32+ 2) /* "rng" RNG present (xstore) */
 #define X86_FEATURE_XSTORE_EN		( 5*32+ 3) /* "rng_en" RNG enabled */
+#define X86_FEATURE_CCS        (5*32+4) /*  "sm3 sm4" present */
```
建议这里和后面没对齐的地方修改缩进和上下文保持一致

opsiff commented:
/ok-to-test

opsiff commented on arch/x86/include/asm/cpufeatures.h:
```diff
@@ -148,6 +148,8 @@
 /* VIA/Cyrix/Centaur-defined CPU features, CPUID level 0xC0000001, word 5 */
 #define X86_FEATURE_XSTORE		( 5*32+ 2) /* "rng" RNG present (xstore) */
 #define X86_FEATURE_XSTORE_EN		( 5*32+ 3) /* "rng_en" RNG enabled */
+#define X86_FEATURE_CCS			(5*32+4) /*  "sm3 sm4" present */
```
这个(5*32+4)相比( 5*32+ 3) 格式不对。下面也需要改。

leoliu-oc replied:
未与文档既定格式保持一致的原因，是因为会导致checkpatch.pl脚本检查不通过。不过，这边会按照建议修改。

leoliu-oc committed:
x86/cpufeatures: Add Zhaoxin feature bits

zhaoxin inclusion
category: feature

-------------------

Add Zhaoxin feature bits on Zhaoxin CPUs.

Signed-off-by: leoliu-oc <leoliu-oc@zhaoxin.com>

leoliu-oc committed:
x86/cpu: Add detect extended topology for Zhaoxin CPUs

zhaoxin inclusion
category: feature

-------------------

Detect the extended topology information of Zhaoxin CPUs if available.

Signed-off-by: leoliu-oc <leoliu-oc@zhaoxin.com>

opsiff commented:
/lgtm
